R/load_met.R

```diff
@@ -0,0 +1,110 @@
+source("./R/convert_to_vera_met_P1D.R")
+
+#' load_met
+#' 
+#' Loads historical and forecasted met data for a given site.
+#'
+#' @param forecast_date reference date for forecast generation
+#' @param forecast_days days into the future to forecast
+#' @param site site
+#'
+#' @return no return. Exports historical and future met
+#'
+load_met <- function(site,
+                     forecast_date,
+                     forecast_days = 35) {
+  
+  message("Loading met data for site ", site)
+  
+  #Stop if too many sites
+  if(length(site) > 1) {
+    stop("length(site) > 1. Only one site can be loaded at a time.")
+  }
+  
+  #Specify variables
+  variables <- c("relativehumidity_2m", 
+                 "precipitation", 
+                 "windspeed_10m", 
+                 "temperature_2m")
+  
+  variables_renamed <- c("RH_percent_mean", 
+                         "Rain_mm_sum", 
+                         "WindSpeed_ms_mean", 
+                         "AirTemp_C_mean")
+  
+  #Load sites
+  site_list <- read_csv("https://raw.githubusercontent.com/LTREB-reservoirs/vera4cast/main/vera4cast_field_site_metadata.csv", 
+                        show_col_types = FALSE)
+  lat <- site_list$latitude[site_list$site_id == site]
+  long <-  site_list$longitude[site_list$site_id == site]
+  if(!site %in% site_list$site_id){
+    stop("Site not found in site list")
+  }
+  
+  #Weather predictions
+  message("Loading weather predictions")
+  weather_pred <- RopenMeteo::get_ensemble_forecast(
+    latitude = lat,
+    longitude = long,
+    forecast_days = forecast_days, # days into the future
+    past_days = 92, # past days that can be used for model fitting
+    model = "gfs_seamless", # this is the NOAA gefs ensemble model
```

Comment:
I would add `model` to the arguement so they can download a different open_meteo models